source/dub/commandline.d

```diff
@@ -1209,8 +1209,10 @@ class GenerateCommand : PackageBuildCommand {
 		gensettings.runArgs = app_args;
 
 		logDiagnostic("Generating using %s", m_generator);
-		dub.generateProject(m_generator, gensettings);
-		if (this.baseSettings.buildType == "ddox") dub.runDdox(gensettings.run, app_args);
+		if (this.baseSettings.buildType == "ddox")
```

Comment:
the log line above may not be accurate anymore if we don't use m_generator, no?

Also it looks like m_generator is checked if it exists and set to the first free arg of the command line, so the user probably expects that generator to run.

The only time it looks like runDdox calls generateProject the same was as here is when calling `dub generate --build=ddox build` - I think we should probably check that m_generator == "build" in this if?

Reply:
Since the CI fails it looks like I was wrong. I'll look more into it soon-ish.